web/models/pair_retro.ex

```diff
@@ -48,15 +52,20 @@ defmodule Pairmotron.PairRetro do
   The update changeset does not allow the user to change the user or pair
   associated with the pair_retro.
   """
-  def update_changeset(struct, params \\ %{}, pair_start_date) do
+  def update_changeset(struct, params \\ %{}, pair, project) do
     struct
     |> cast(params, @required_update_fields, @optional_fields)
     |> foreign_key_constraint(:project_id)
-    |> validate_field_is_not_before_date(:pair_date, pair_start_date)
+    |> validate_date_is_not_before_pair(:pair_date, pair)
     |> validate_field_is_not_in_future(:pair_date)
+    |> validate_project_is_for_group(:project_id, project, pair)
   end
 
-  defp validate_field_is_not_before_date(changeset, field, pair_start_date) do
+  defp validate_date_is_not_before_pair(changeset, field, pair) do
+    pair_start_date = cond do
+      is_nil(pair) -> nil
+      true -> Pairmotron.Calendar.first_date_of_week(pair.year, pair.week)
```

Comment:
This could be two functions if you pattern match on pair == nil. Looks like it might simplify logic below in the changeset validation.